src/Cake.Npm/NpmTool.cs

```diff
@@ -93,6 +93,26 @@ protected void RunCore(TSettings settings, ProcessSettings processSettings, Acti
             processSettings.RedirectStandardError = settings.RedirectStandardError;
             processSettings.RedirectStandardOutput = settings.RedirectStandardOutput;
 
+            if(settings.StandardOutputAction != null)
+            {
+                processSettings.RedirectStandardOutput = true;
+                processSettings.RedirectedStandardOutputHandler = x =>
+                {
+                    settings.StandardOutputAction(x);
```

Comment:
Any specific reason to only provide an action and not a function like `RedirectedStandardOutputHandler`?

Reply:
Well, yes. Since the `Redirect...Handlers` are in fact used to [modify the output](https://github.com/cake-build/cake/blob/develop/src/Cake.Core/IO/ProcessWrapper.cs#L63) before it is used, I opted for not allowing modifications of the output because modifying the StandardOutput might break the parsing of the output in `NpmVersion` and `NpmPack`.